torch_xla/csrc/ops/nll_loss2d.cpp

```diff
@@ -0,0 +1,80 @@
+#include "torch_xla/csrc/ops/nll_loss2d.h"
+
+#include "absl/types/span.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/nll_loss.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& logits, const Value& labels,
+                           const absl::optional<Value>& weight,
+                           ReductionMode reduction, int ignore_index) {
+  auto lower_for_shape_fn =
+      [&](absl::Span<const xla::XlaOp> operands) -> xla::XlaOp {
+    absl::optional<xla::XlaOp> weight;
```

Comment:
There wouldn't be need of using optionals as XlaOp already has the ability to identify itself as invalid:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/xla_builder.h#L79

But I guess I missed that in a code review, so it is OK to leave it as such for now.

Reply:
Ah, I see. I will remove the usage in this pr and submit another one to clean up the rest of the optional<xla::XlaOp>.